server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowTaskInstanceLoggerService.java

```diff
@@ -0,0 +1,128 @@
+        Optional<TaskEntity> taskEntityOptional =
+                flowTaskInstanceService.getLogDownloadableTaskEntity(flowInstanceId, false);
+        return getLogContent(taskEntityOptional, level, flowInstanceId);
+    }
+
+    @SneakyThrows
+    public String getLogContentWithoutPermission(OdcTaskLogLevel level, Long flowInstanceId) {
+        Optional<TaskEntity> taskEntityOptional =
+                flowTaskInstanceService.getLogDownloadableTaskEntity(flowInstanceId, true);
+        return getLogContent(taskEntityOptional, level, flowInstanceId);
+    }
+
+    @SneakyThrows
+    private String getLogContent(Optional<TaskEntity> taskEntityOptional, OdcTaskLogLevel level, Long flowInstanceId) {
+        if (!taskEntityOptional.isPresent()) {
+            log.warn("get log failed, flowInstanceId={}", flowInstanceId);
+            return LogUtils.DEFAULT_LOG_CONTENT;
+        }
+        TaskEntity taskEntity = taskEntityOptional.get();
+        if (!dispatchChecker.isTaskEntityOnThisMachine(taskEntity)) {
+            ExecutorInfo executorInfo = JsonUtils.fromJson(taskEntity.getExecutor(), ExecutorInfo.class);
+            DispatchResponse response = requestDispatcher.forward(executorInfo.getHost(), executorInfo.getPort());
+            return response.getContentByType(new TypeReference<SuccessResponse<String>>() {}).getData();
+        }
+        return getLog(taskEntity.getCreatorId(), taskEntity.getId() + "", taskEntity.getTaskType(), level);
+    }
+
+    @SneakyThrows
+    public File downloadLog(Long flowInstanceId) {
+        Optional<TaskEntity> taskEntityOptional =
+                flowTaskInstanceService.getLogDownloadableTaskEntity(flowInstanceId, false);
+        TaskEntity taskEntity = taskEntityOptional
+                .orElseThrow(() -> new NotFoundException(ErrorCodes.NotFound, new Object[] {flowInstanceId},
+                        ErrorCodes.TaskLogNotFound.getLocalizedMessage(new Object[] {"Id", flowInstanceId})));
+        if (!dispatchChecker.isTaskEntityOnThisMachine(taskEntity)) {
+            ExecutorInfo executorInfo = JsonUtils.fromJson(taskEntity.getExecutor(), ExecutorInfo.class);
+            DispatchResponse response = requestDispatcher.forward(executorInfo.getHost(), executorInfo.getPort());
+            return response.getContentByType(new TypeReference<SuccessResponse<File>>() {}).getData();
+        }
+        return getLogFile(taskEntity.getCreatorId(), taskEntity.getId() + "", taskEntity.getTaskType(),
+                OdcTaskLogLevel.ALL);
+    }
+
+    public File getLogFile(Long userId, String flowInstanceId, TaskType type, OdcTaskLogLevel logLevel) {
+        String logFilePath = taskService.getLogFilePath(userId, flowInstanceId, type, logLevel);
+        try {
+            return taskService.getLogFile(logFilePath);
+        } catch (NotFoundException ex) {
+            log.warn(ErrorCodes.TaskLogNotFound.getEnglishMessage(new Object[] {"Id", flowInstanceId}));
```

Comment:
log.warn("Task log file not found, flowInstanceId={}, logFilePath={}", flowInstanceId, logFilePath);

Reply:
ok